### PR TITLE
Adding Java samples for listUsersAsync() and setCustomUserClaimsAsync()

### PR DESCRIPTION
**Pull request by hiranya911**
Related to https://github.com/firebase/firebase-admin-java/pull/92

Samples to appear on https://firebase.google.com/docs/auth/admin/custom-claims and https://firebase.google.com/docs/auth/admin/manage-users

### CODE REVIEW

File: auth/src/main/java/com/google/firebase/quickstart/AuthSnippets.java [hunk 1]
```diff
@@ -91,6 +93,81 @@ public static void updateUser(String uid) throws InterruptedException, Execution
...
+    FirebaseAuth.getInstance().setCustomClaimsAsync(uid, claims).get();
+    // The new custom claims will propagate to the user's ID token the
+    // next time a new one is issued.
+    // [END set_custom_user_claims]
+
+    String idToken = "id_token";
+    // [START verify_custom_claims]
+    // Verify the ID token first.
+    FirebaseToken decoded = FirebaseAuth.getInstance().verifyIdTokenAsync(idToken).get();
+    if (Boolean.TRUE.equals(decoded.getClaims().get("admin"))) {
```
Comment (samtstern) [inline auth/src/main/java/com/google/firebase/quickstart/AuthSnippets.java]: `Boolean.TRUE.equals` seems a bit strange.  Why not just `if(decoded.getClaims().get("admin"))`

File: auth/src/main/java/com/google/firebase/quickstart/AuthSnippets.java [hunk 2]
```diff
@@ -91,6 +93,81 @@ public static void updateUser(String uid) throws InterruptedException, Execution
...
+    // [END set_custom_user_claims_script]
+  }
+
+  public static void setCustomUserClaimsInc() throws InterruptedException, ExecutionException {
+    // [START set_custom_user_claims_incremental]
+    UserRecord user = FirebaseAuth.getInstance()
+        .getUserByEmailAsync("user@admin.example.com").get();
+    // Add incremental custom claim without overwriting the existing claims.
+    Map<String, Object> currentClaims = user.getCustomClaims();
+    if (Boolean.TRUE.equals(currentClaims.get("admin"))) {
```
Comment (samtstern) [inline auth/src/main/java/com/google/firebase/quickstart/AuthSnippets.java]: Same here.

File: auth/src/main/java/com/google/firebase/quickstart/AuthSnippets.java (hunk 1 above)
Comment (hiranya911) [reply auth/src/main/java/com/google/firebase/quickstart/AuthSnippets.java]: `getClaims()` returns a `Map<String, Object>`. So when querying the map we need to deal with possible null values (if the key does not exist), and the returned value not being boolean. `Boolean.TRUE.equals()` provides a one-liner that deals with both cases.

**Comment by samtstern** [comment]
SGTM!

File: auth/src/main/java/com/google/firebase/quickstart/AuthSnippets.java [hunk 3]
```diff
@@ -155,6 +232,8 @@ public static void main(String[] args) throws InterruptedException, ExecutionExc
     getUserByEmail("user@example.com");
     getUserByPhoneNumber("+11234567890");
     updateUser("some-uid");
+    //setCustomUserClaims("some-uid");
```
Comment (avishalom) [inline auth/src/main/java/com/google/firebase/quickstart/AuthSnippets.java]: Is this commented out on purpose ?

**Pull request merged**